pkg/k8s/services.go

```diff
@@ -53,9 +53,10 @@ func BuildKubeClient(kubeconfigPath string) (kubernetes.Interface, error) {
 	return clientset, nil
 }
 
-// GetKubeStateMetricsURL fetches the URL for the Kube State Metrics service
-func GetKubeStateMetricsURL(ctx context.Context, clientset kubernetes.Interface, namespace string) (string, error) {
-	services, err := clientset.CoreV1().Services(namespace).List(ctx, metav1.ListOptions{})
+// GetKubeStateMetricsURL fetches the URL for the Kube State Metrics service across all namespaces
+func GetKubeStateMetricsURL(ctx context.Context, clientset kubernetes.Interface) (string, error) {
+	// First, try to find the service by name
+	services, err := clientset.CoreV1().Services("").List(ctx, metav1.ListOptions{})
```

Comment:
By setting the `namespace` arg to `""`, the function call will return all services across all namespaces.